mbed/mbed.py

```diff
@@ -177,6 +177,22 @@ def progress():
     sys.stdout.flush()
     sys.stdout.write('\b')
 
+def show_progress(title, percent, max_width=80):
+    percent = round(float(percent), 2)
+    show_percent = '%.2f' % percent
+    line = str(title) + ' |'
+    bwidth = max_width - len(str(title)) - len(show_percent) - 6
```

Comment:
This `6` seems a litte magic. Is it `len(' |') + len('| ') + len('%') + len('\r')`?

Reply:
It's the ones you mentioned + an extra character as on some terminals (like Windows) printing exactly 80 characters, produces a new line.